### DISCUSSION TITLE
How would I go about modifying this code for a Teensy 4.1 from pjrc.com ?

### DISCUSSION THREAD

**Discussion by radiohound** [pschatzmann/arduino-espeak-ng#5] [Ideas]
Hello,
I would like to modify your code to allow a Teensy 4.1 with arduino-espeak-ng to work. There is a sound library and example for the Teensy that allows for 22khz sound files to be played. It is located here: https://github.com/FrankBoesing/Teensy-WavePlayer/blob/main/examples/WavFilePlayerI2SSampleRate/WavFilePlayerI2SSampleRate.ino , (a fork of the well documented original library located here: https://www.pjrc.com/teensy/td_libs_Audio.html ). The specifications on the Teensy 4.1 are 8192 kB Flash and 1023 kB memory with a M7 processor at 600 mhz. So it is quite powerful and can do many sound related tasks. It also has a handy design tool for adding filters, sound components, etc https://www.pjrc.com/teensy/gui/index.html . Those are a few reasons I am really interested in using this hardware.

Do you have some suggestions how I could go about doing this?

**Comment by pschatzmann** [comment]
I don't think that you need to modify anything.
Just have a look at the included example and adapt it to your purpose. You define the output by passing a Print or Stream object (in the constructor). In your case this would be e.g. AudioOutputI2S, but I don't know the Teesy API good enough to be of any help
